src/CaptureClient/SaveToFileCaptureEventProcessor.cpp

```diff
@@ -0,0 +1,117 @@
+// Copyright (c) 2021 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "CaptureClient/CaptureEventProcessor.h"
+#include "CaptureFile/CaptureFile.h"
+#include "CaptureFile/CaptureFileOutputStream.h"
+#include "OrbitBase/MakeUniqueForOverwrite.h"
+#include "capture_data.pb.h"
+
+using orbit_capture_file::CaptureFile;
+using orbit_capture_file::CaptureFileOutputStream;
+using orbit_client_protos::UserDefinedCaptureInfo;
+using orbit_grpc_protos::ClientCaptureEvent;
+
+namespace orbit_capture_client {
+
+namespace {
+
+class SaveToFileCaptureEventProcessor : public CaptureEventProcessor {
+ public:
+  explicit SaveToFileCaptureEventProcessor(std::filesystem::path file_path,
+                                      absl::flat_hash_set<uint64_t> frame_track_function_ids,
+                                      std::function<void(const ErrorMessage&)> error_handler)
+      : frame_track_function_ids_(std::move(frame_track_function_ids)),
+        file_path_{std::move(file_path)},
+        error_handler_{std::move(error_handler)} {}
```

Comment:
Should Initialize be called in the constructor?

Reply:
Constructors can't return ErrorMessageOr<>, this is one of the reasons we have a factory method here.